### PR TITLE
Add support for passive service healthchecks

### PR DESCRIPTION
**Pull request by simu**
Replacement for #4

### CODE REVIEW

File: gc/gc.go [hunk 1]
```diff
@@ -57,14 +74,19 @@ func Collect(ts time.Time, c config.Configuration) error {
 		l.Errorf(fmt.Sprintf("[Collect] Error while listing services: %v", err))
 		return err
 	}
+	downtimes, err := icinga.ListDowntimes(fmt.Sprintf("host=%v", hostname))
```
Comment (corvus-ch) [inline gc/gc.go]: This should probably go after the log on line 82.

File: webhook/icinga_test.go [hunk 2]
```diff
@@ -67,3 +69,87 @@ func TestComputeServiceName(t *testing.T) {
...
+				"exitStatus": "1",
+				"interval":   "330",
+				"message":    "the message 2",
+			},
+		},
+	}
+	c := config.NewMockConfiguration(1)
+	i := icinga2.NewMockClient()
+
+	for idx, alert := range alerts {
```
[the remark this answers is not in the document]
Comment (simu) [reply webhook/icinga_test.go]: I've refactored the tests, can you have a quick look again

File: webhook/icinga.go [hunk 3]
```diff
@@ -139,6 +141,29 @@ func updateOrCreateService(icinga icinga2.Client,
...
+		// periods are already managed by Prometheus/Alertmanager
+		MaxCheckAttempts: 1,
+	}
+
+	// Check if this is a heartbeat service. Adjust serviceData
+	// accordingly
+	if val, ok := alert.Labels["heartbeat"]; ok {
+		heartbeat_interval, err := time.ParseDuration(val)
+		if err != nil {
+			return icinga2.Service{}, fmt.Errorf("Unable to parse heartbeat interval: %v", err)
```
[the remark this answers is not in the document]
Comment (simu) [reply webhook/icinga.go]: a struct cannot be `nil`, and the method does not return a pointer-to-struct

File: webhook/icinga_test.go [hunk 4]
```diff
@@ -67,3 +69,87 @@ func TestComputeServiceName(t *testing.T) {
 		assert.Equal(t, err == nil, expected, fmt.Sprintf("Alert: %+v -> %v; err = %v", alert, svcName, err))
 	}
 }
+
+func TestUpdateOrCreateService(t *testing.T) {
+	alerts := []template.Alert{
```
[the remark this answers is not in the document]
Comment (simu) [reply webhook/icinga_test.go]: refactored accordingly

File: webhook/icinga_test.go [hunk 5]
```diff
@@ -67,3 +69,87 @@ func TestComputeServiceName(t *testing.T) {
...
+	for idx, alert := range alerts {
+		svcName, err := computeServiceName(template.Data{}, alert, c)
+		displayName, err := computeDisplayName(template.Data{}, alert)
+		svc, err := updateOrCreateService(i, "test.vshn.net",
+			svcName, displayName, alert, c)
+		assert.Equal(t, err == nil, true, fmt.Sprintf("Alert: %+v -> %v; err = %v", alert, svc, err))
+		assert.Equal(t, svc.MaxCheckAttempts == 1, true, "soft states disabled for check %v", displayName)
+		isHeartbeat, _ := strconv.ParseBool(alert.Annotations["heartbeat"])
+		var state, check string
+		if isHeartbeat {
```
[the remark this answers is not in the document]
Comment (simu) [reply webhook/icinga_test.go]: refactored accordingly

**Pull request merged**